src/main/java/week5/AlphabetFinderTest.java

```diff
@@ -0,0 +1,102 @@
+        is(Arrays.asList('A', 'T', 'R', 'C')));
+  }
+
+  @Test
+  public void testContradiction() {
+    //does the program crash on contradiction
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "ARC", "CAT", "CAR"}),
+        is(Arrays.asList('A', 'T', 'R', 'C')));
+  }
+
+  @Test
+  public void testMergeCorrection() {
+    //partial order refines alphabet - there is a letter that should be moved forward
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "CTT", "CTA", "CRA"}),
+        is(Arrays.asList('T', 'A', 'R', 'C')));
+  }
+
+  @Test
+  public void testMergeCorrectionB() {
+    //partial order refines alphabet - there is a letter that should be moved backward
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"AC", "DR", "D", "TAR", "TAC", "TAD"}),
+        is(Arrays.asList('A', 'R', 'C', 'D', 'T')));
+  }
+
+  @Test
+  public void testAmbiguous() {
+    //partial order is more ambigious than alphabet
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "CAT", "CAA", "DT", "DC"}),
+        is(Arrays.asList('T', 'A', 'R', 'C', 'D')));
+  }
+
+  @Test
+  public void testCasing() {
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "cAT", "CAR", "CAa"}),
+        is(Arrays.asList('A', 'R', 'c', 'C', 'T', 'a')));
+  }
+
+  @Test
+  public void testDuplication() {
+    //duplicate words
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "RA", "RAT", "CAT", "CAR"}),
+        is(Arrays.asList('A', 'T', 'R', 'C')));
+  }
+
+  @Test
+  public void testAllLetter() {
+
+    //making sure all letters are included in the alphabet
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "CAT", "CAR", "CAD"}),
+        is(Arrays.asList('A', 'T', 'R', 'C', 'D')));
+  }
+
+  @Test
+  public void testDiffLen() {
+    //in this example 'D' is before 'S' because 'D' is encountered on the third
+    //character of the word whereas 'S' on the fourth
+    assertThat(
+        AlphabetFinder.findAlphabet(new String[]{"ARTS", "RATTA", "CAT", "CAR", "CADDDDDDDDDD"}),
+        is(Arrays.asList('A', 'T', 'R', 'C', 'D', 'S')));
+  }
+
+  @Test
+  public void testMultipleCorrections() {
+    //in this example 'D' is before 'S' because 'D' is encountered on the third
+    //character of the word whereas 'S' on the fourth
+    assertThat(
+        AlphabetFinder.findAlphabet(new String[]{"ARTS", "RATTA", "CAT", "CAR", "CADDDDDDDDDD"}),
+        is(Arrays.asList('A', 'T', 'R', 'C', 'D', 'S')));
+  }
+
+  @Test
+  public void testMultipleCorrections2() {
+    //multiple letters needs to move forward due to a partial order
+    // e.g. alphabet = [ A, R, D, T, C, B] and po = [C, B, R] -> alphabet to [A, C, B, R, D, T]
+    assertThat(
+        AlphabetFinder
+            .findAlphabet(new String[]{"ART", "RB", "RD", "RT", "TC", "T", "SC", "SB", "SA"}),
+        is(Arrays.asList('C', 'B', 'A', 'R', 'D', 'T', 'S')));
+  }
+}
```

Comment:
It's a decent-looking selection of tests - I particularly like the tests on duplicates and mixed case, for example.

However, it has missed bugs in your code. If I use the input {"AH", "BG", "CF", "DF", "DG", "EG", "EH"}, it returns [A, B, C, D, E, G, H, F].